examples/grpc-server-with-prometheus/client/client.go

```diff
@@ -22,13 +42,35 @@ func main() {
 
 	defer conn.Close()
 
+	// Create a HTTP server for prometheus.
+	httpServer := &http.Server{Handler: promhttp.HandlerFor(reg, promhttp.HandlerOpts{}), Addr: fmt.Sprintf("0.0.0.0:%d", 9094)}
+
+	// Start your http server for prometheus.
+	go func() {
+		if err := httpServer.ListenAndServe(); err != nil {
+			log.Fatal("Unable to start a http server.")
+		}
+	}()
+
 	// Create a gRPC server client.
 	client := pb.NewDemoServiceClient(conn)
-	// Call “SayHello” method and wait for response from gRPC Server.
-	resp, err := client.SayHello(context.Background(), &pb.HelloRequest{Name: "Test"})
-	if err != nil {
-		log.Fatal(err)
+	fmt.Println("Start to call the method called SayHello every 3 seconds")
+	go func() {
+		for {
+			// Call “SayHello” method and wait for response from gRPC Server.
+			_, err := client.SayHello(context.Background(), &pb.HelloRequest{Name: "Test"})
+			if err != nil {
+				log.Fatal(err)
```

Comment:
We might need better error handling than this. I don't think log.Fatal works as expected in go routine (https://stackoverflow.com/questions/34884570/golang-why-os-exit-doesnt-work-inside-goroutines). Same for httpServer.ListenAndServer err handling. I think we might want just error log instead?

Reply:
If you really want nice error handling we might want to use this https://github.com/oklog/run

Reply:
Can we add some extra packages to this example?

Reply:
About the statement 「I don't think log.Fatal works as expected in go routine」.
Actually, I learnt knowledge about scheduler of Golang.  I found that the exception perhaps happened in the program which constantly runs a loop and appends goroutines to runnable queue. The main reason of it is that the scheduler can not get a change to run other goroutines. But, in this PR,  we have three goroutines, `httpServer`, `SayHello`, `main`, which run individualy in this program. The scheduler of Golang can schedule them without any mistake. So, I don't think that we need to add extra package to handle a uncertain problem about scheduling in Golang. Most importantly, this example should be as simple as possible.

Reply:
sure, but I would advice (especially for examples) to never use panics / os.Exit(1) unless it is main function. I might be fine with ListenAndServe `Fatal` log, but I why we cannot just print error for simplicity?

Reply:
Sure. I will probably remove `log.Fatal` from the calling sayHello method.